jpsonic-main/src/main/java/com/tesshu/jpsonic/dao/AlbumDao.java

```diff
@@ -306,6 +306,20 @@ where present and folder_id in (:folders) and year between :toYear and :fromYear
         }
     }
 
+    public List<Album> getAlbumsByGenre(int offset, int count, List<String> genres, List<MusicFolder> folders) {
+        if (genres.isEmpty() || folders.isEmpty()) {
+            return Collections.emptyList();
+        }
+        Map<String, Object> args = LegacyMap.of("folders", MusicFolder.toIdList(folders), "genres", genres, "count",
+                count, "offset", offset);
+        return template.namedQuery("select " + QUERY_COLUMNS + """
+                from album
+                where present and folder_id in (:folders) and genre in (:genres)
+                order by album_order
+                limit :count offset :offset
+                """, rowMapper, args);
+    }
+
```

Comment:
Speed ​​Considerations
1) Consider adding an index to the genre field. If it has a negative effect, it will not be added.
2) Move to com.tesshu.jpsonic.dao.dialect.
 - Searching with Lucene is faster. (Only id searches are used in the database. However, when using the in clause, overflow may occur depending on the use of the database, so it is assumed that unnest is used. In any case, the database is limited, so it is treated as dialect.)

Reply:
Done. Adding an index will be considered during speed verification.